types/backbone.marionette/index.d.ts

```diff
@@ -1311,6 +1311,11 @@ export interface CollectionViewOptions<
      */
     filter?(child?: TModel, index?: number, collection?: TCollection): boolean;
 
+    /**
+     * Prevent some of the underlying children from being attached to the DOM.
+     */
+    viewFilter?: ((view?: typeof Backbone.View, index?: number, children?: Backbone.View[]) => boolean) | Backbone.ObjectHash | string;
```

Comment:
Thanks to @reubenrybnik, you might be able to use [`_.Iteratee`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/1e69551ff518516feb8aa9ea04e11c2eda95a82a/types/underscore/index.d.ts#L105-L111) for this:

```suggestion
    viewFilter?: _.Iteratee<Backbone.View[], boolean>;
```

I'm not sure whether this type is actually visible outside of that module, so it might not work, but if it does, it seems preferable.

Reply:
I'm not familiar with this code, so anything I say should definitely be taken with a grain of salt, but I just wanted to make a few quick notes here about this potential change:

- Depending on whether or not this supports all the different iteratee types of Underscore functions, `_.CollectionIterator` may be a better fit here than `_.Iteratee`
- For either `_.CollectionIterator` or `_.Iteratee` to work well here, the types for `view` and `index` would need to be non-optional (I think `Iteratee<Backbone.View[] | undefined, boolean>` would at least allow for a possibly undefined `children` parameter though)
- For either `CollectionIterator` or `Iteratee` to work well here, the type for the `view` parameter would need to extend `TypeOfCollection<Backbone.View[]>, and `typeof Backbone.View` (which is the type of the base Backbone view's constructor function rather than the base type of an instantiated view) does not do that

Reply:
Too, I guess I'd say that unless the underlying implementation for this is defined more or less as a proxy for an Underscore function it might be better to not reference Underscore's types here since if Underscore later changes its types in a way that is not compatible with this function it may lead to either incorrect typings for this function or extra work untangling the types.

Reply:
Types from underscore are already used extensively here (especially in the `Container` type), so that shouldn't be an issue. But I've tried to use the suggested Iteratee and CollectionIterator but I couldn't make it work. Maybe someone with more typescript-fu can get it to work.